cinder/scheduler/manager.py

```diff
@@ -237,19 +241,33 @@ def create_snapshot(self, ctxt, volume, snapshot, backend,
         """
         self._wait_for_scheduler()
 
+        if CONF.sap_allow_independent_snapshots:
+            # We allow snapshots to be created on a pool
+            # separate from the volume's pool.
+            backend = volume_utils.extract_host(volume['host'])
+
         try:
             tgt_backend = self.driver.backend_passes_filters(
```

Comment:
Is it here? Or does this only check validity?

Reply:
the scheduler does it with the backend value only containing the cinder backend specified.

The cinder host is structured by
host@backend_name#pool

when the scheduler is passed host@backend_name#pool, then it is required to match the destination of host@backend_name#pool.

When the scheduler is passed host@backend_name then it can pick any pool on that backend.

So this change tells the scheduler to pick any pool on the same backend as the source volume.